zebra-scan/src/service/scan_task/scan.rs

```diff
@@ -325,36 +354,26 @@ pub async fn scan_height_and_store_results(
         // TODO: use the real sapling tree size: `zs::Response::SaplingTree().position() + 1`
         let sapling_tree_size = 1 << 16;
 
-        tokio::task::spawn_blocking(move || {
-            let dfvk_res =
-                scan_block(&network, &block, sapling_tree_size, &dfvks).map_err(|e| eyre!(e))?;
-            let ivk_res =
-                scan_block(&network, &block, sapling_tree_size, &ivks).map_err(|e| eyre!(e))?;
-
-            let dfvk_res = scanned_block_to_db_result(dfvk_res);
-            let ivk_res = scanned_block_to_db_result(ivk_res);
-
-            let latest_subscribed_keys = subscribed_keys_receiver.borrow().clone();
-            if let Some(results_sender) = latest_subscribed_keys.get(&sapling_key).cloned() {
-                let results = dfvk_res.iter().chain(ivk_res.iter());
-
-                for (_tx_index, &tx_id) in results {
-                    // TODO: Handle `SendErrors` by dropping sender from `subscribed_keys`
-                    let _ = results_sender.try_send(ScanResult {
-                        key: sapling_key.clone(),
-                        height,
-                        tx_id: tx_id.into(),
-                    });
-                }
-            }
+        // TODO: Next code should run in a new thread but it seems we need to implement `Send` for upstream `ScanningKeys` to do so.
 
-            storage.add_sapling_results(&sapling_key, height, dfvk_res);
-            storage.add_sapling_results(&sapling_key, height, ivk_res);
+        let dfvk_res = scan_block(&network, &block, sapling_tree_size, ready_scan_block_keys)
```

Comment:
We are now only scanning once hoping that the new key types will also include the `ivk`s however i am not totally sure about that. Need to confirm.

Reply:
@upbqdn can you check this one? it is outdated by other changes but still relevant. Thanks!

Reply:
I somehow missed this.

We're now using `ScanningKeys` which use `ivk` internally: https://github.com/zcash/librustzcash/blob/99dfbb9202d9ff58f6e508ed986219f0036b4c14/zcash_client_backend/src/scanning.rs#L269-L270.

Reply:
Yea, this is old stuff, thank you.